google/cloud/storage/tests/object_insert_integration_test.cc

```diff
@@ -602,10 +620,40 @@ TEST_P(ObjectInsertIntegrationTest, InsertWithUserIpBlank) {
   ASSERT_STATUS_OK(insert_meta);
   ScheduleForDelete(*insert_meta);
 
-  EXPECT_THAT(log.ExtractLines(),
-              Contains(AllOf(HasSubstr(" POST "),
-                             HasSubstr("/b/" + bucket_name_ + "/o"),
-                             HasSubstr("userIp="))));
+  EXPECT_THAT(
+      log.ExtractLines(),
+      Contains(
+          AllOf(HasSubstr(" POST "), HasSubstr("/b/" + bucket_name_ + "/o"),
+                MatchesRegex(R"regex(.*userIp=([0-9]+\.){3}[0-9]+.*)regex"))));
```

Comment:
If you used `ContainsRegex()` instead, you wouldn't need the `.*`s required by anchoring (and below).